skforecast/utils/utils.py

```diff
@@ -1513,25 +1513,46 @@ def exog_to_direct_numpy(
 
 def expand_index(
     index: Union[pd.Index, None], 
-    steps: int
-) -> pd.Index:
+    steps: Union[int, str, pd.Timestamp],
+    **kwargs
+) -> Tuple[pd.Index, int]:
     """
     Create a new index of length `steps` starting at the end of the index.
     
     Parameters
     ----------
     index : pandas Index, None
         Original index.
-    steps : int
-        Number of steps to expand.
+    steps : Union[int, str, pd.Timestamp]
+        Number of steps to expand (if type is int) or
+        Datetime until the index is expanded (str formatted as YYYY-MM-DD or pd.Timestamp)
```

Comment:
I will change the docstring to:

If `steps` is int, number of steps to expand. If str or pandas Datetime, expand index to this date (str must be convertible to pandas Datetime format).